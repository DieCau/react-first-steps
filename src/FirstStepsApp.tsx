import { ItemCounter } from './shopping-cart/ItemCounter';

interface ItemInCart {
  productName: string;
  quantity: number;
}

const itemInCart: ItemInCart[] = [
  { productName: 'Nintendo Switch 2', quantity: 2 },
  { productName: 'Mario Bros 3', quantity: 5 },
  { productName: 'Play 5', quantity: 3 },
];

export function FirstStepsApp() {
  return (
    <>
      <h1>Carrito de Compras</h1>

      {itemInCart.map(({ productName, quantity }) => (
        <ItemCounter
          key={productName}
          productName={productName}
          quantity={quantity}
        />
      ))}

      {/* <ItemCounter productName='Nintendo Switch 2' quantity={5} />
      <ItemCounter productName='Play 5' quantity={3} />
      <ItemCounter productName='Attari 2000' quantity={6} />
      <ItemCounter productName='Super Mario 2' quantity={4} /> */}
    </>
  );
}
