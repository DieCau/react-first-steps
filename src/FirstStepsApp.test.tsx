import { describe, expect, test } from 'vitest';
import { FirstStepsApp } from './FirstStepsApp';
import { render } from '@testing-library/react';

describe('FirstStepApp', () => {
  test('should match snapshot', () => {
    const { container } = render(<FirstStepsApp />);
    expect(container).toMatchSnapshot();
  });
});
