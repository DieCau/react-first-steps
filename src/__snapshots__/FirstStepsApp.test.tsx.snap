// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FirstStepApp > should match snapshot 1`] = `
<div>
  <h1>
    Carrito de Compras
  </h1>
  <section
    class="_itemRow_8b32a2"
  >
    <span
      class="_item-text_8b32a2"
      style="color: black;"
    >
      Nintendo Switch 2
    </span>
    <button>
      +1
    </button>
    <span>
      2
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="_itemRow_8b32a2"
  >
    <span
      class="_item-text_8b32a2"
      style="color: black;"
    >
      Mario Bros 3
    </span>
    <button>
      +1
    </button>
    <span>
      5
    </span>
    <button>
      -1
    </button>
  </section>
  <section
    class="_itemRow_8b32a2"
  >
    <span
      class="_item-text_8b32a2"
      style="color: black;"
    >
      Play 5
    </span>
    <button>
      +1
    </button>
    <span>
      3
    </span>
    <button>
      -1
    </button>
  </section>
</div>
`;
